compiler/src/compile.re

```diff
@@ -177,9 +177,15 @@ let rec compile_resume = (~hook=?, s: compilation_state) => {
   };
 };
 
+let reset_between_compiles = () => {
```

Comment:
I might call this `reset_compiler_state`, but otherwise looks great!

Reply:
Done